lib/src/screens/receive/lightning_invoice_page.dart

```diff
@@ -0,0 +1,346 @@
+  final FocusNode _amountFocusNode;
+
+  final LightningViewModel lightningViewModel;
+  final LightningInvoicePageViewModel lightningInvoicePageViewModel;
+  final ReceiveOptionViewModel receiveOptionViewModel;
+  final _formKey = GlobalKey<FormState>();
+
+  bool effectsInstalled = false;
+
+  @override
+  bool get gradientAll => true;
+
+  @override
+  bool get resizeToAvoidBottomInset => false;
+
+  @override
+  bool get extendBodyBehindAppBar => true;
+
+  @override
+  AppBarStyle get appBarStyle => AppBarStyle.transparent;
+
+  @override
+  void onClose(BuildContext context) => Navigator.popUntil(context, (route) => route.isFirst);
+
+  @override
+  Widget middle(BuildContext context) => PresentReceiveOptionPicker(
+      receiveOptionViewModel: receiveOptionViewModel, color: titleColor(context));
+
+  @override
+  Widget trailing(BuildContext context) => TrailButton(
+      caption: S.of(context).clear,
+      onPressed: () {
+        _formKey.currentState?.reset();
+      });
+
+  Future<bool> _onNavigateBack(BuildContext context) async {
+    onClose(context);
+    return false;
+  }
+
+  @override
+  Widget body(BuildContext context) {
+    WidgetsBinding.instance.addPostFrameCallback((_) => _setReactions(context));
+
+    return WillPopScope(
+      onWillPop: () => _onNavigateBack(context),
+      child: KeyboardActions(
+        disableScroll: true,
+        config: KeyboardActionsConfig(
+            keyboardActionsPlatform: KeyboardActionsPlatform.IOS,
+            keyboardBarColor: Theme.of(context).extension<KeyboardTheme>()!.keyboardBarColor,
+            nextFocus: false,
+            actions: [
+              KeyboardActionsItem(
+                focusNode: _amountFocusNode,
+                toolbarButtons: [(_) => KeyboardDoneButton()],
+              ),
+            ]),
+        child: Container(
+          color: Theme.of(context).colorScheme.background,
+          child: ScrollableWithBottomSection(
+            contentPadding: EdgeInsets.only(bottom: 24),
+            content: Container(
+              decoration: responsiveLayoutUtil.shouldRenderMobileUI
+                  ? BoxDecoration(
+                      borderRadius: BorderRadius.only(
+                          bottomLeft: Radius.circular(24), bottomRight: Radius.circular(24)),
+                      gradient: LinearGradient(
+                        colors: [
+                          Theme.of(context)
+                              .extension<ExchangePageTheme>()!
+                              .firstGradientTopPanelColor,
+                          Theme.of(context)
+                              .extension<ExchangePageTheme>()!
+                              .secondGradientTopPanelColor,
+                        ],
+                        begin: Alignment.topLeft,
+                        end: Alignment.bottomRight,
+                      ),
+                    )
+                  : null,
+              child: Observer(builder: (_) {
```

Comment:
**Minor:**
is this observer needed? please make sure its behaving correctly without it first, but from what I see it's not used